src/utils/focus.js

```diff
@@ -0,0 +1,49 @@
+const TAB_KEY = 9;
+const trapFocusHandlers = {};
+
+export function trapFocus(container) {
+  const focusableElements = container.querySelectorAll("a, button, input, select");
+  const firstElement = focusableElements[0];
+  const lastElement = focusableElements[focusableElements.length - 1];
+
+  removeTrapFocus(container);
+
+  trapFocusHandlers.focusin = function (event) {
+    if (event.target !== firstElement && event.target !== lastElement) { 
+      return;
+    }
+
+    container.addEventListener("keydown", trapFocusHandlers.keydown);
+  };
+
+  trapFocusHandlers.focusout = function () {
+    container.removeEventListener("keydown", trapFocusHandlers.keydown);
+  };
+
+  trapFocusHandlers.keydown = function (event) {
+    if (event.keyCode !== TAB_KEY) { 
+      return;
+    }
+
+    if (event.target === lastElement && !event.shiftKey) {
+      event.preventDefault();
+      firstElement.focus();
+    }
+
+    if (event.target === firstElement && event.shiftKey) {
+      event.preventDefault();
+      lastElement.focus();
+    }
+  };
+
+  container.addEventListener("focusout", trapFocusHandlers.focusout);
+  container.addEventListener("focusin", trapFocusHandlers.focusin);
+
+  firstElement.focus();
+}
+
+export function removeTrapFocus(container) {
```

Comment:
Just a nit but since this function is called in `trapFocus`, does it make sense to move this up top?

Also, this is probably more defensive than it needs to be but if `trapFocusHandlers` is empty, we technically don't need to run the following code

Reply:
I can add a check before each `removeEventListener` to make sure that `trapFocusHandlers` contains the listener function that is being removed
```
if (trapFocusHandlers.focusin) {
  container.removeEventListener("focusin", trapFocusHandlers.focusin);
} 
```